pkg/cloudprovider/providers/azure/azure_backoff.go

```diff
@@ -131,7 +80,12 @@ func (az *Cloud) CreateOrUpdateSGWithRetry(sg network.SecurityGroup) error {
 		resp := <-respChan
 		err := <-errChan
 		glog.V(10).Infof("SecurityGroupsClient.CreateOrUpdate(%s): end", *sg.Name)
-		return processRetryResponse(resp.Response, err)
+		done, err := processRetryResponse(resp.Response, err)
+		if done && err == nil {
+			// Update the cache right after updating.
+			az.nsgCache.AddOrUpdate(*sg.Name, &sg)
```

Comment:
Any reason 2nd parameter not nil only here? how is it different from lb/vm?

Reply:
NSG rule won't change after POST to ARM，while VM/LB may be changed (e.g. allocate a private IP for LB)